libkbfs/mdserver_remote.go

```diff
@@ -351,6 +351,10 @@ func (md *MDServerRemote) OnDisconnected(ctx context.Context,
 		md.serverOffset = 0
 	}()
 
+	md.authenticatedMtx.Lock()
```

Comment:
I was following your lead above in `resetAuth` to keep it consistent -- that approach is fine with me when the line has no chance of panicking.